src/app/server/Server.cpp

```diff
@@ -237,7 +231,35 @@ CHIP_ERROR Server::Init(AppDelegate * delegate, uint16_t secureServicePort, uint
     SuccessOrExit(err);
 
     err = mCASESessionManager.Init();
-
+#if !CHIP_DEVICE_CONFIG_ENABLE_THREAD
```

Comment:
Please document why this does not need to happen if ENABLE_THREAD.

Especially for a device that supports both Thread and WiFi or Ethernet and is not on the Thread network.

(Followup ok for sorting out what the actual conditions here should be.)

Reply:
I added a comment on why it does not need to happen if ENABLE_THREAD is defined, but a follow up is necessary on a per platform basis by the manifacturer's to validate they can rejoin groups after a reboot.